docs/Authentication.md

```diff
@@ -0,0 +1,327 @@
+# Authentication for Azure MCP Server
+
+This document provides comprehensive authentication guidance for Azure MCP Server, with a focus on advanced scenarios including protected resources, firewall restrictions, and enterprise environments.
```

Comment:
It looks to me a doc of security best practices rather than authentications.

e.g. IMO, firewall restrictions are not authentication related. "Protected Resources with Local Auth Disabled" is not part of "authentication guidance" either.